### PR TITLE
Sling 11331: resource resolver: events need to be queued until vanity path loading is complete

### PR DESCRIPTION
**Pull request by reschke**

### CODE REVIEW

File: src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java [hunk 1]
```diff
@@ -139,6 +143,8 @@ public class MapEntries implements
 
     private byte[] vanityBloomFilter;
 
+    private boolean vanityPathsProcessed;
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java:146]: I think this should be volatile, or maybe use AtomicBoolean

File: src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java [hunk 2]
```diff
@@ -714,6 +783,17 @@ public void onChange(final List<ResourceChange> changes) {
                 continue;
             }
 
+            // during startup: just enqueue the events
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java:786]: It's maybe better to do the check at method start and then handle all changes in the same way. If the check is done here, the first changes might go into the queue, while the remaining ones are not.

Comment (reschke) [reply src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java:786]: that could only happen when "vanityPathsProcessed" ever changes from true to false, right?

Comment (cziegeler) [reply src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java:789]: right

**Review by cziegeler — APPROVED** [review]
Lgtm., thanks

**Pull request merged**